src/amo/components/SearchContextCard/index.js

```diff
@@ -117,15 +179,72 @@ export class SearchContextCardBase extends React.Component {
   }
 }
 
-export function mapStateToProps(state) {
+export function mapStateToProps(state: AppState) {
+  const { search } = state;
+  const { filters } = search;
+
+  let currentCategory;
+
+  if (
+    search &&
+    filters &&
+    filters.category &&
+    typeof filters.category === 'string'
+  ) {
+    currentCategory = filters.category;
+  }
+
+  let categoryName;
+
+  if (currentCategory && !categoryName) {
+    const categoriesState = state.categories.categories;
+    const { clientApp } = state.api;
+    const allCategories = [];
+
+    if (categoriesState && clientApp) {
+      Object.keys(categoriesState[clientApp]).forEach((type) => {
+        const searchType = categoriesState[clientApp][type];
+        Object.keys(searchType).forEach((category) => {
+          const searchCategory = searchType[category];
+          const { slug } = searchCategory;
+          const { name } = searchCategory;
+          allCategories.push({
+            [slug]: {
+              name,
+              slug,
+            },
+          });
+        });
+      });
+    }
+
+    const translatedCategory =
+      allCategories.length &&
+      allCategories.find(
+        (category) =>
+          category[currentCategory] &&
+          category[currentCategory].slug === currentCategory,
+      );
+
+    categoryName =
+      translatedCategory && translatedCategory[currentCategory]
+        ? translatedCategory[currentCategory].name
+        : null;
+  }
+
   return {
-    count: state.search.count,
-    filters: state.search.filters,
-    loading: state.search.loading,
+    hasCategory: !!currentCategory,
+    categoryName,
+    count: search.count,
+    filters,
+    loading: search.loading,
```

Comment:
Shouldn't this be the `loading` value from the `categories` state, not the `search` state, or do you perhaps need both? After looking at the tests, it seems like at least some of the time you are concerned with whether the `categories` are loading, but it looks like other times you might be concerned with whether the search results are loading.

Reply:
yeh maybe you're right (I need both). I will look into this further with fresh 👀

Reply:
When there is a category, then we do need to worry about category but I think we can assume when we have categoryName, categories have loaded so I *think* this is okay.

I think because search depends on the category info (if it's present), then I think this is okay to use this for loading here. It is confusing but does that make sense?

Reply:
I also removed the test I added `should not fetch categories if there is a category filter but it's already loading` since I am not going to using the loading check any longer. If `categoryName` is present, I am using this to determine if we should do the dispatch. I think this is the right way to handle this but let me know if you don't agree :/

Reply:
I've looked over the lastest patch and I think you're right - you don't need loading status for categories, just for search. It might make sense to change the prop name from `loading` to `loadingSearch` to that it's clear which loading status we're concerned with.